tla-io/src/main/scala/at/forsyte/apalache/tla/imp/SanyImporter.scala

```diff
@@ -25,16 +25,31 @@ class SanyImporter(sourceStore: SourceStore, annotationStore: AnnotationStore) {
    *
    * @param file
    *   an input file
+   * @param useLibraryPaths
+   *   Whether to use [[file]]'s containing directory as library path. Disable this flag when you [[loadFromSource]]; we
+   *   then look up names in the paths given by the `TLA-Library` Java system variable instead.
    * @return
    *   the pair (the root module name, a map of modules)
    */
-  def loadFromFile(file: File): (String, Map[String, TlaModule]) = {
+  def loadFromFile(file: File, useLibraryPaths: Boolean = true): (String, Map[String, TlaModule]) = {
     // create a string buffer to write SANY's error messages
-    val errBuf = new StringWriter()
     // use.toString() to retrieve the error messages
-    val specObj =
-      new SpecObj(file.getAbsolutePath, new SanyNameToStream())
+    val errBuf = new StringWriter()
+
+    // Resolver for filenames, patched for wired modules.
+    // If `file` has a parent directory and `useLibraryPaths` is set, look up files in this parent directory.
+    // Otherwise, look up files in the paths given by the `TLA-Library` Java system variable (used to wire in the
+    // Apalache Standard library in `src/tla` when running unit tests).
+    // Compatible with TLC at commit https://github.com/tlaplus/tlaplus/commit/b905a86e31714bf5002290141b51b41ffe280c8e
+    val filenameResolver = if (useLibraryPaths) {
+      file.getParent() match {
+        case null          => SanyNameToStream()
```

Comment:
Is it a fallback case? I guess, the only case in Linux/MacOS is when the user passes `/` as the filename.

Reply:
For the rare case of a non-hierarchical file system? In any case, `getParent()`

> Returns [...] null if this pathname does not name a parent directory.

so we have to handle it somehow.

Reply:
I just realized that `getParent()` also returns `null` on a relative path without parent directory.
This is called in 3 different places, so I'm calling `getAbsoluteFile()` in 168c3d88 below to make sure this is an absolute path.